jetty-core/jetty-websocket/jetty-websocket-jetty-tests/src/test/java/org/eclipse/jetty/websocket/tests/ExplicitDemandTest.java

```diff
@@ -114,4 +137,27 @@ public void testNoDemandWhenProcessingFrame() throws Exception
         assertNull(clientSocket.error);
         assertNull(serverSocket.error);
     }
+
+    @Test
+    public void testNoAutoDemand() throws Exception
+    {
+        URI uri = new URI("ws://localhost:" + connector.getLocalPort() + "/listenerSocket");
+        ListenerSocket listenerSocket = new ListenerSocket();
+        Future<Session> connect = client.connect(listenerSocket, uri);
+        Session session = connect.get(5, TimeUnit.SECONDS);
+
+        session.sendPing(ByteBuffer.wrap("ping-0".getBytes(StandardCharsets.UTF_8)), Callback.NOOP);
```

Comment:
Add a test case where the server sends a PING, so the client receives a PING, and it should be able to receive subsequent frames.